app/Traits/AccessControlAPI.php

```diff
@@ -140,16 +140,18 @@ function () use ($api_key, $fileset_hash, $access_group_ids) {
 
                 $dbp_database = config('database.connections.dbp.database');
 
+                $is_bileis_api_key =  $this->isBibleIsApiKey($api_key);
+
                 return AccessGroupKey::join('user_keys', function ($join) use ($api_key) {
                     $join->on('user_keys.id', '=', 'access_group_api_keys.key_id')
                         ->where('user_keys.key', $api_key);
                 })->join(
                     $dbp_database . '.access_group_filesets as acc_filesets',
-                    function ($join) use ($fileset_hash, $access_group_ids) {
+                    function ($join) use ($fileset_hash, $access_group_ids, $is_bileis_api_key) {
                         $join->on('access_group_api_keys.access_group_id', '=', 'acc_filesets.access_group_id')
                             ->where('hash_id', $fileset_hash);
 
-                        if (!empty($access_group_ids)) {
+                        if (!empty($access_group_ids) && $is_bileis_api_key === false) {
```

Comment:
This isn't where I was expecting the change. In my view, the change should be in method allowedForDownload.
something like this
            download_access_group_list = access groups associated with bibleis key
        } else {
            // current implementation
            $download_access_group_list = config('settings.download_access_group_list');
        }
... same

Reply:
@bradflood It has changed the old logic by:

```php
if ($this->doesApiKeyBelongToBibleis($this->key)) {
    $download_access_group_array_ids = AccessGroupKey::getAccessGroupIdsByApiKey($this->key)->toArray();
} else {
    $download_access_group_array_ids = explode(',', config('settings.download_access_group_list'));
}
```